### PR TITLE
BZ 1318344 - BPMN Process with swimlane created from business-centra has been distorted while opening in JBDS

### PR DESCRIPTION
**Pull request by tsurdilo**
@jlindop running the test is enough for the test (no need to check in eclipse unless you feel like it)

thanks.

### CODE REVIEW

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/bpmn2/impl/Bpmn2UnmarshallingTest.java [hunk 1]
```diff
@@ -1016,4 +1020,25 @@ public void testDocumentationForSwimlane() throws Exception {
         assertEquals("Swimlane name is wrong.", lane.getName(), "Documented Swimlane");
         assertEquals("<![CDATA[Some documentation for swimlane.]]>", lane.getDocumentation().get(0).getText());
     }
+
+    @Test
+    public void testSequenceFlowPointsInsideLane() throws Exception {
+        Bpmn2JsonUnmarshaller unmarshaller = new Bpmn2JsonUnmarshaller();
```
Comment (jomarko) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/bpmn2/impl/Bpmn2UnmarshallingTest.java]: Hello @tsurdilo , yesterday there was merged my commit which simplifies initialization of each testcase. Definitions can be now loaded from json in the following way:
`Definitions definitions = loader.loadProcessFromJson("sequenceFlowPointsInsideLane.json");`

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java [hunk 2]
```diff
@@ -246,6 +246,32 @@ public void revisitDI(Definitions def) {
         revisitEdgeBoundsInContainers(def);
     }
 
+    public Bounds getShapeBoundsForElement(BaseElement fn, BPMNPlane plane) {
```
Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java]: Please do not add this method. Such method already exists on master branch:

https://github.com/droolsjbpm/jbpm-designer/blob/master/jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java#L583

Feel free to rename the existing method above if you feel that it doesn't describe itself enough.

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java [hunk 3]
```diff
@@ -2911,29 +2954,30 @@ private void createSubProcessDiagram(BPMNPlane plane, FlowElement flowElement, B
...
-				}
-				point = dcFactory.createPoint();
-				if(sequenceFlow.getTargetRef() != null) {
-					Bounds targetBounds = _bounds.get(sequenceFlow.getTargetRef().getId());
-					point.setX(targetBounds.getX() + (targetBounds.getWidth()/2));
-					point.setY(targetBounds.getY() + (targetBounds.getHeight()/2));
-				}
-				edge.getWaypoint().add(point);
-				plane.getPlaneElement().add(edge);
+                SequenceFlow sequenceFlow = (SequenceFlow) subProcessFlowElement;
```
Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java]: For better readability I propose to align this block of code in standard way:
`
if (...) {
    // any code here
} else if (...) {
   // any code here
}
`

Comment (jomarko) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java]: For better readability I propose to align this block of code in standard way. For example:
https://github.com/droolsjbpm/jbpm-designer/blob/master/jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonUnmarshaller.java#L2954-2977

**Comment by tsurdilo** [comment]
@jomarko Thanks for the suggestions. The problem I have with this however is that the master branch currently does not run on wildfly and the way we have to do is test with 6.4.x and then cherry-pick to master. Your suggestions are not in 6.4.x so this stuff can happen. 

If I make these changes I will not be able to re-test this vs master currently so I guess we merge this as is (as its tested and your suggestions are something we can revise later as well) or don't merge at all until I am able to run master on 6.4.x . 
@krisv any suggestions?

**Comment by psiroky** [comment]
On master we will support only WildFly 10+. I would not invest the time to make it run in WildFly 8, but rather in WildFly 10.

**Comment by jomarko** [comment]
@tsurdilo yes, my changes are not on 6.4.x branch because @csadilek announced me in #227 that 6.4.x branch is in blockers only mode. So after #227 I stopped creating PR's for 6.4.x branch if I wanted contribute only with refactoring or adding tests. I thought that blockers only mode means that only fixes for BZ's with flag blocker can be merged into 6.4.x branch. Maybe I understand blockers only mode bad.

**Comment by tsurdilo** [comment]
@jomarko no problem. i will run on wildfly 10+ and update this pr . thanks @psiroky !

**Comment by tsurdilo** [comment]
i will redo this PR

**Pull request closed without merging**